src/main/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/graphs/JTBestHaplotype.java

```diff
@@ -0,0 +1,133 @@
+package org.broadinstitute.hellbender.tools.walkers.haplotypecaller.graphs;
+
+import org.broadinstitute.hellbender.tools.walkers.haplotypecaller.readthreading.ExperimentalReadThreadingGraph;
+import java.util.*;
+import java.util.stream.Collectors;
+
+/**
+ * A best haplotype object for being used with junction trees.
+ *
+ * Each path holds a list of all the junction trees describing its current path. This list consists of pointers to nodes
+ * in the junction trees corresponding to the paths that have already been taken by the JTBestHaplotype object.
+ *
+ * In order to invoke the junction trees simply call {@link #getApplicableNextEdgesBasedOnJunctionTrees} which will return a list
+ * of cloned path objects corresponding to each path present in the eldest tree. This method handles popping old trees with insufficient
+ * data off of the list as well as incrementing all of the trees in the list to point at the next element based on the chosen path.
+ */
+public class JTBestHaplotype<T extends BaseVertex, E extends BaseEdge> extends KBestHaplotype<T, E> {
+    // TODO these nodes probably shouldn't be exposed and held externally methinks
+    private List<ExperimentalReadThreadingGraph.ThreadingNode> activeNodes;
+
+    public JTBestHaplotype(final JTBestHaplotype p, final List<E> edgesToExtend, final double edgePenalty) {
+        super(p, edgesToExtend, edgePenalty);
+        //this.threadingTreesToConsult = p.threadingTreesToConsult.cop;
+        activeNodes = new ArrayList<ExperimentalReadThreadingGraph.ThreadingNode>(p.activeNodes);
+    }
+
+    // Constructor to be used for internal calls from {@link #getApplicableNextEdgesBasedOnJunctionTrees()}
+    public JTBestHaplotype(final JTBestHaplotype p, final List<E> chain, final int edgeMultiplicity, final int totalOutgoingMultiplicity) {
+        super(p, chain, computeLogPenaltyScore( edgeMultiplicity, totalOutgoingMultiplicity));
+        activeNodes = new ArrayList<ExperimentalReadThreadingGraph.ThreadingNode>(p.activeNodes);
+        // Ensure that the relevant edge has been traversed
+        takeEdge(chain.get(chain.size() - 1));
+    }
+
+    public JTBestHaplotype(final T initialVertex, final BaseGraph<T,E> graph) {
+        super(initialVertex, graph);
+        activeNodes = new ArrayList<>();
+    }
+
+    //TODO this needs to be the same logic as the blow method, this is temporary
+    // returns true if there is a symbolic edge pointing to the reference end or if there is insufficient node data
+    public boolean hasStoppingEvidence(final int weightThreshold) {
+        ExperimentalReadThreadingGraph.ThreadingNode eldestTree = activeNodes.isEmpty() ? null : activeNodes.get(0);
```

Comment:
oldest is better than eldest here

Reply:
But we are following this tree because it is so wizened and knowledgable